spec/unit/provider/package/gem_spec.rb

```diff
@@ -36,6 +40,34 @@
         allow(provider).to receive(:rubygem_version).and_return('1.9.9')
       end
 
+      context 'on windows' do
+        let(:path) do
+          "C:\\Program Files\\Puppet Labs\\Puppet\\puppet\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\Ruby26-x64\\bin;C:\\Windows\\system32\\bin"
+        end
+
+        let(:expected_path) do
+          "C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\Ruby26-x64\\bin;C:\\Windows\\system32\\bin"
+        end
+
+        before do
+          allow(Puppet::Util::Platform).to receive(:windows?).and_return(true)
+          allow(Puppet::Util).to receive(:get_env)
+          allow(Puppet::Util).to receive(:get_env).with('PATH').and_return(path)
+          allow(described_class).to receive(:validate_command).with(provider_gem_cmd)
+          stub_const('::File::PATH_SEPARATOR', ';')
```

Comment:
We'll need to restore this when done. Maybe easier to just confine the `on windows` context to `if: Puppet::Util::Platform.windows?` instead of trying to run it everywhere.

Reply:
I don't really like the idea of having unit tests behaving differently depending on the platform they are run.

[stub_const](https://relishapp.com/rspec/rspec-mocks/v/2-11/docs/stubbing-constants) will restore to the original value after the example is run so no need to take care of that.

```ruby

  context 'when stubbing' do
    before do
      stub_const('::File::PATH_SEPARATOR', ';')
    end
    it 'is ;' do
      puts "::File::PATH_SEPARATOR => #{::File::PATH_SEPARATOR}"
      expect(::File::PATH_SEPARATOR).to eql(';')
    end
  end

  context 'after stubbing' do
    it do
      puts "::File::PATH_SEPARATOR => #{::File::PATH_SEPARATOR}"
      expect(::File::PATH_SEPARATOR).not_to eql(';')
    end
  end
end
```

```
description
  when stubbing
::File::PATH_SEPARATOR => ;
    is ;
  after stubbing
::File::PATH_SEPARATOR => :
    is expected not to eql ";"
```

Reply:
> I don't really like the idea of having unit tests behaving differently depending on the platform they are run.

True, but the behavior *is* different on windows vs others, so testing the windows behavior on non-windows platforms isn't super useful. You'd have to also stub `::File::ALT_SEPARATOR` and `::File::SEPARATOR` in order to more closely mimic Windows (otherwise `C:\\Program Files\\Puppet Labs\\Puppet\\puppet\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\Ruby26-x64\\bin;C:\\Windows\\system32\\bin"` would appear to be a relative path for a single file).

Reply:
> stub_const will restore to the original ...

My bad, I misread that. I thought it said `const_set`

Reply:
> isn't super useful

That said I'm fine to merge as is